SConstruct

```diff
@@ -180,6 +172,24 @@ opts.Add(
 
 opts.Add(BoolVariable("build_library", "Build the godot-cpp library.", True))
 
+# CPU architecture options.
+architecture_array = ["auto", "universal", "x86_32", "x86_64", "arm32", "arm64", "rv64", "ppc32", "ppc64", "wasm32"]
```

Comment:
This should be synced with the godot repo.

There we removed `universal` as an arch because it's not an actual architecture and can cause bugs when checking `env["arch"] == "x86_64"` for example. We normally build for macOS by building `x86_64` and `arm64` specifically (so each build can use the right options for that arch) and `lipo` the results.

Similarly, in the godot repo `auto` is not part of the list. It's added only for the option but the list itself in `platform_methods.py` is reused to do validation.

Reply:
I don't like the idea of removing `universal` as an option. For the engine, it's more complicated and it makes sense to require explicitly `lipo`-ing to make a universal binary. For GDNative, the use cases are vastly simpler and it makes sense to me to provide an easy option. Plus, `universal` was already an option before.

Aside from keeping universal, I have synced this list, removing `auto`.